app/src/main/java/com/example/sharewhatyoucanproject/login/OnBoardingFragment.kt

```diff
@@ -0,0 +1,47 @@
+package com.example.sharewhatyoucanproject.login
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.appcompat.app.AppCompatActivity
+import androidx.fragment.app.Fragment
+import androidx.navigation.fragment.findNavController
+import com.example.sharewhatyoucanproject.databinding.FragmentOnBoardingBinding
+import com.example.sharewhatyoucanproject.models.UserType
+import com.google.firebase.auth.FirebaseAuth
+
+class OnBoardingFragment : Fragment() {
+
+    private var _binding: FragmentOnBoardingBinding? = null
+    private val binding get() = _binding!!
+
+    override fun onCreate(savedInstanceState: Bundle?)
+    {
+        super.onCreate(savedInstanceState)
+        FirebaseAuth.getInstance().signOut()
```

Comment:
can you wrap this in a utility method or within a view model so that you are not using firebase auth directly from a fragment?

A common pattern in android is to create Repository classes that wrap database and network libraries. Scroll down to the "Data Layer" section in the docs: https://developer.android.com/topic/architecture

Reply:
wrapped in view model